src/GafferSceneUI/LightTool.cpp

```diff
@@ -760,495 +792,438 @@ class LightToolHandle : public Handle
+		// Adds an inspection for all metaParameters for the current context.
+		void addInspection()
+		{
+			InspectionMap inspectionMap;
+			for( const auto &m : m_metaParameters )
+			{
+				InspectionInfo i;
+				if( inspectionInfo( m, i ) )
+				{
+					inspectionMap[m] = i;
+				}
+			}
 
-		const std::string m_lightTypePattern;
+			m_inspections.push_back( inspectionMap );
+		}
 
-		Gaffer::PlugPtr m_editScope;
+		void clearInspections()
+		{
+			m_inspections.clear();
+		}
 
-		bool m_lookThroughLight;
-};
+		bool handleDragMove( const GafferUI::DragDropEvent &event )
+		{
+			if( m_inspections.empty() || !allInspectionsEnabled() )
+			{
+				return true;
+			}
 
-// ============================================================================
-// SpotLightHandle
-// ============================================================================
+			return handleDragMoveInternal( event );
+		}
 
-class SpotLightHandle : public LightToolHandle
-{
+		bool handleDragEnd()
+		{
+			m_dragStartInspection.clear();
 
-	private :
+			return handleDragEndInternal();
+		}
 
-		// A struct holding the angle inspections and the original angles during a drag.
-		// Angles are in "handle-space" (generally 1/2 the full cone for the cone angle
-		// and the full penumbra angle for penumbras. See `handleAngles` and `plugAngles`
-		// for conversion details.)
-		struct DragStartData
+		void updateLocalTransform( const V3f &scale, const V3f &shear )
 		{
-			Inspector::ResultPtr coneInspection;
-			float originalConeHandleAngle;
-			Inspector::ResultPtr penumbraInspection;
-			std::optional<float> originalPenumbraHandleAngle;
-		};
-
-	public :
+			updateLocalTransformInternal( scale, shear );
+		}
 
-		enum class HandleType
+		bool enabled() const
 		{
-			Cone,
-			Penumbra
-		};
+			// Return true without checking the `enabled()` state of our inspections.
+			// This allows the tooltip-on-highlight behavior to show a tooltip explaining
+			// why an edit is not possible. The alternative is to draw the tooltip for all
+			// handles regardless of mouse position because a handle can only be in a disabled
+			// or highlighted drawing state.
+			// The drawing code takes care of graying out uneditable handles and the inspections
+			// prevent the value from being changed.
+			return !m_inspectors.empty();
+		}
 
-		SpotLightHandle(
-			const std::string &lightType,
-			HandleType handleType,
-			const SceneView *view,
-			const float zRotation,
-			const std::string &name = "SpotLightHandle"
-		) :
-			LightToolHandle( lightType, name ),
-			m_view( view ),
-			m_zRotation( zRotation ),
-			m_handleType( handleType ),
-			m_angleMultiplier( 1.f ),
-			m_visualiserScale( 1.f ),
-			m_frustumScale( 1.f ),
-			m_lensRadius( 0 ),
-			m_dragStartData {}
+		// Derived classes can optionally extend behavior by first calling the base class method
+		// and performing more tests if needed to determine visibility for the `scenePath` in the
+		// current context.
```

Comment:
This documentation seems inaccurate, since `visible()` isn't virtual.. Perhaps the public `visible()` function should be documented for public consumption, and the documentation about derived class overrides should go on the `visibleInternal()` method?

Reply:
I tried making things a bit clearer in https://github.com/GafferHQ/gaffer/pull/5483/commits/c12d27e2b943cda1243559cd1483e9a61a317ed2 and https://github.com/GafferHQ/gaffer/pull/5483/commits/98d314c1bbc7aaa560a05047f1f80757f312d7be